osquery/main/main.cpp

```diff
@@ -178,6 +182,13 @@ int startOsquery(int argc, char* argv[]) {
 
   // Only worker processes should start a daemon or shell.
   if (!runner.isWatcher()) {
+#if defined(__unix__) && !defined(__APPLE__)
+    auto status = configureBPFMemoryLimits();
```

Comment:
Could this instead be called within the BPF publishers? For example each publisher could call this configure API and within this method you could implement a `std::call_once` to set the RLIMIT.